express-notes-api/index.js

```diff
@@ -0,0 +1,92 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+const fs = require('fs');
+
+const notes = data.notes;
+
+app.get('/api/notes', (req, res) => {
+  const notesArray = [];
+  for (const id in notes) {
+    notesArray.push(notes[id]);
+  }
+  res.status(200).json(notesArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = Number(req.params.id);
+  if (id < 0) {
+    res.status(400).json({ error: 'Please input a positive integer for id' });
+  } else if (notes[id]) {
+    res.status(200).json(notes[id]);
+  } else if (!notes[id]) {
+    res.status(404).json({ error: 'Cannot find note with id ' + id });
+  }
+});
+
+app.use(express.json());
+
+let id = data.nextId;
+
+app.post('/api/notes', (req, res) => {
+  if (!req.body.content) {
+    res.status(400).json(({ error: 'Content is a required field' }));
+  } else if (req.body.content !== undefined) {
+    const dataJSON = JSON.stringify(data, null, 2);
+    fs.writeFile('data.json', dataJSON, 'utf8', function (err) {
+      if (err) {
+        console.error(err);
+        res.status(500).json({ error: 'An unexpected error has occurred' });
+      } else {
+        notes[id] = req.body;
+        notes[id].id = id;
+        id++;
```

Comment:
The new data needs to be added to the data model _before_ you stringify and write it. Your creations, updates, and deletes should persist to the data.json file.

Reply:
Noted! Will post those revisions asap